File: iterator/EmployeeHierarchyCollection.java
package iterator;

import java.util.List;

public class EmployeeHierarchyCollection implements IterableCollection {

    private final List<Employee> employees;
    private final String employeeId;

    public EmployeeHierarchyCollection(String employeeId) {
        this.employees = List.of(new Employee("123", "Mahmoud"), new Employee("345", "Ahmed"));
        this.employeeId = employeeId;
    }

    public List<Employee> getEmployees() {
        return employees;
    }

    @Override
    public EmployeeHierarchyIterator createEmployeeDirectReportIterator() {
        return new EmployeeDirectReportIterator(this);
    }

    @Override
    public EmployeeHierarchyIterator createEmployeeCoWorkerIterator() {
        return new EmployeeCoWorkerIterator(this);
    }

    @Override
    public EmployeeHierarchyIterator createEmployeeSubOrdinateIterator() {
        return new EmployeeSubOrdinateIterator(this);
    }
}

File: PHP/Patterns/src/Behavioral/Iterator/EmployeeHierarchyCollection.php
<?php

namespace src\Behavioral\Iterator;

class EmployeeHierarchyCollection implements IteratableCollection
{
    private array $employees;
    private $employeeId;

    /**
     * @param array $employees
     * @param $employeeId
     */
    public function __construct($employeeId)
    {
        $this->employees = [
            new Employee(123, "Ahmed"),
            new Employee(456, "Ali"),
            new Employee(789, "Aya"),
            new Employee(234, "Nada"),
            new Employee(678, "Mohammed"),
        ];

        $this->employeeId = $employeeId;
    }

    public function getEmployees(): array
    {
        return $this->employees;
    }


    function createEmployeeDirectReportIterator(): EmployeeHierarchyIterator
    {
        return new EmployeeDirectReportIterator($this);
    }

    function createEmployeeCoWorkerIterator(): EmployeeHierarchyIterator
    {
        return new EmployeeCoWorkerIterator($this);
    }

    function createEmployeeSubordinateIterator(): EmployeeHierarchyIterator
    {
        return new EmployeeSubOrdinateIterator($this);
    }
}